### PR TITLE
Apple iMac 14,2: init

### PR DESCRIPTION
**Pull request by drupol**
###### Things done

- [x] Tested the changes in your own NixOS Configuration
- [x] Tested the changes end-to-end by using your fork of `nixos-hardware` and
      importing it via `<nixos-hardware>` or Flake input

This PR requires https://github.com/NixOS/nixpkgs/pull/337948 to work correctly.

### CODE REVIEW

File: apple/imac/14-2/default.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  ];
+
+  boot = {
+    initrd.availableKernelModules = [
+      "nvme"
+      "sd_mod"
+      "thunderbolt"
+      "usb_storage"
+      "xhci_pci"
+    ];
```
Comment (Mic92) [inline apple/imac/14-2/default.nix:22]: We do not include code that can is generated by nixos-generate-config in nixos-hardware

```suggestion
```

File: apple/imac/14-2/default.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      "brcmfmac"
+      "brcmsmac"
+      "bcma"
+    ];
+    kernelPackages = lib.mkIf (lib.versionOlder pkgs.linux.version "6.0") pkgs.linuxPackages_latest;
+    extraModulePackages = [ config.boot.kernelPackages.broadcom_sta ];
+  };
+
+  hardware = {
+    enableRedistributableFirmware = lib.mkDefault true;
```
Comment (Mic92) [inline apple/imac/14-2/default.nix:42]: Already the default in NixOS:

```suggestion
```

File: apple/imac/14-2/default.nix [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    cpu.intel.updateMicrocode = lib.mkDefault true;
```
Comment (Mic92) [inline apple/imac/14-2/default.nix:43]: Part of the intel cpu profile:

```suggestion
```

File: apple/imac/14-2/default.nix [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  lib,
+  pkgs,
+  ...
+}:
+{
+  imports = [
+    ../.
+    ../../../common/gpu/intel
+    ../../../common/gpu/nvidia
+    ../../../common/hidpi.nix
```
Comment (Mic92) [inline apple/imac/14-2/default.nix:12]: I believe we got rid of this?

```suggestion
```

Comment (drupol) [reply apple/imac/14-2/default.nix:12]: No it's still there.

Comment (Mic92) [reply apple/imac/14-2/default.nix:12]: Ah. We just removed quite a bunch of stuff.

**Comment by drupol** [comment]
@Mic92 Ready for final review now.

**Comment by Mic92** [comment]
@mergify queue

**Pull request merged**